src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -1842,6 +1841,56 @@ class CreateStatementSegment(BaseSegment):
     )
 
 
+@snowflake_dialect.segment(replace=True)
+class CreateViewStatementSegment(BaseSegment):
+    """A `CREATE VIEW` statement, specifically for Snowflake's dialect.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-view.html
+    """
+
+    type = "create_view_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        Ref("OrReplaceGrammar", optional=True),
+        AnyNumberOf(
+            "SECURE",
+            "RECURSIVE",
+        ),
+        "VIEW",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("ObjectReferenceSegment"),
+        AnyNumberOf(
+            Sequence(
+                Bracketed(
+                    Delimited(
+                        Sequence(
+                            Ref("ObjectReferenceSegment"),
+                            Ref("CommentClauseSegment", optional=True),
+                        ),
+                    ),
+                ),
+            ),
+            Sequence(
+                Sequence("WITH", optional=True),
```

Comment:
```suggestion
                Ref.keyword("WITH", optional=True),
```

Reply:
For my education: When should we use `Sequence(...)` vs `Ref.keyword(...)`?

Reply:
So you can think of `Ref.keyword` as the explicit way to refer to a keyword segment. Usually you don't need to specify it and can just write the keyword e.g. `"WITH"` but it's useful when you want to specify additional parameters e.g. an optional WITH keyword: `Ref.keyword("WITH", optional=True)`.

`Sequence` is a sequence of multiple segments e.g. you could specify an optional group of segments `Sequence("IF", "EXISTS", optional=True)` and the entire sequence is optional, you can't have one of the segments without the others.

It's not completely wrong to use `Sequence` here, it just makes more sense in this case to read it as an optional WITH rather than an optional sequence of length 1 😄